packages/backend/src/tokens/handshake.ts

```diff
@@ -134,13 +134,14 @@ export class HandshakeService {
     }
 
     const redirectUrl = this.removeDevBrowserFromURL(this.authenticateContext.clerkUrl);
-    const frontendApiNoProtocol = this.authenticateContext.frontendApi.replace(/http(s)?:\/\//, '');
 
-    const baseUrl = this.authenticateContext.proxyUrl
-      ? this.authenticateContext.proxyUrl.replace(/\/$/, '')
-      : `https://${frontendApiNoProtocol}`;
+    let baseUrl = this.authenticateContext.frontendApi.startsWith('http')
```

Comment:
Is this check to guard against the developer providing their proxy URL with a protocol?